pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -64,18 +84,158 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 	reqLogger.Info("Reconciling UserAccount")
 
 	// Fetch the UserAccount instance
-	instance := &toolchainv1alpha1.UserAccount{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	userAcc := &toolchainv1alpha1.UserAccount{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: config.GetOperatorNamespace(), Name: request.Name}, userAcc)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
 			return reconcile.Result{}, nil
 		}
-		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, nil
+	var created bool
+	var user *userv1.User
+	if user, created, err = r.ensureUser(userAcc); err != nil {
+		_ = r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, err.Error())
+		return reconcile.Result{}, err
+	} else if created {
+		return reconcile.Result{}, r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, "")
+	}
+
+	var identity *userv1.Identity
+	if identity, created, err = r.ensureIdentity(userAcc); err != nil {
+		_ = r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, err.Error())
```

Comment:
same with the status here and also in case of mapping

Reply:
btw, either we return the error and handle it correctly or, do not return the error, but definitely not ignore it - in this case, you could log it

Reply:
Inside updateStatus() err is logged if not nil.

The idea is, if caller already has one_error (like error in create_identity) then he will use that error and fail `reconcile()`.  If there are no_error and if updateStatus() failed then use that error and fail `reconcile()`.

Reply:
> Inside updateStatus() err is logged if not nil.

that's the point, I think that the error should be logged outside of this method. 
Anyway, every golinter would scream on ignoring returned errors and I would call it also as a good practice - to handle returned errors...

Reply:
btw, as for the logging (I didn't noticed it during the review) - we should use the logger with the additional values https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller.go#L83 and not the "global variable". The reason is that the values give us a context which resource the error/info is related to

Reply:
> that's the point, I think that the error should be logged outside of this method.

@MatousJobanek anyway @alexeykazakov has created follow-up PR for this.  Here is my thought on this.

I have thought all possible way before making this change.  I understand that log statement should be outside but for that I need to check "err != nil" for each call made to `updateStatus()` and as updateStatus is called many time (now and usage will increase in future code) I see that the code will be having lot of this `err != nil` check all over and it will **compromise the readability of code**.  So even though linter will scream I was fine as gaining readability of code.

Also, this method is really internal/private to this controller so I was fine adding log.err inside method.  If this is public then obviously it should return the result and left on caller to log/not_log it.